identity-apps-core/apps/authentication-portal/src/main/webapp/login.jsp

```diff
@@ -1069,6 +1072,56 @@
                                     if (registeredLocalAuthenticators.contains(localAuthenticator)) {
                                         continue;
                                     }
+
+                                    if (localAuthenticator.startsWith(CUSTOM_LOCAL_AUTHENTICATOR_PREFIX)) {
+                                       
+                                        String customLocalAuthenticatorImageURL = "libs/themes/default/assets/images/authenticators/custom-authenticator.svg";
+                                        String customLocalAuthenticatorDisplayName = localAuthenticator;
+                                        Map<String, String> authenticatorConfigMap = new HashMap<>();
+                                        try {
+                                            AuthenticatorDataRetrievalClient authenticatorDataRetrievalClient = new AuthenticatorDataRetrievalClient();
+                                            authenticatorConfigMap = authenticatorDataRetrievalClient.getAuthenticatorConfig(tenantDomain, localAuthenticator);
+                                        } catch (AuthenticatorDataRetrievalClientException e) {
+                                            // Exception is ignored and the default values will be used as a fallback.
+                                        }
+
+                                        if (MapUtils.isNotEmpty(authenticatorConfigMap) && authenticatorConfigMap.containsKey("definedBy") 
+                                            && authenticatorConfigMap.get("definedBy").equals("USER")) {
+                                            
+                                            if (authenticatorConfigMap.containsKey("image")) {
+                                                customLocalAuthenticatorImageURL = authenticatorConfigMap.get("image");
+                                            }
+                                            customLocalAuthenticatorDisplayName = authenticatorConfigMap.get("displayName");
+                            %>
+                                <div class="social-login blurring social-dimmer">
+                                    <div class="field">
+                                            <button
+                                                type="button"
+                                                id="icon-<%=iconId%>"
+                                                class="ui button secondary"
+                                                data-testid="login-page-sign-in-with-custom-local-authenticator<%=localAuthenticator%>"
```

Comment:
Since the authenticator name will already contain `custom`, shall we construct the test id as follows?

```suggestion
                                                data-testid="login-page-sign-in-with-<%=localAuthenticator%>"
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/7544/commits/b795637e8db0cdf9ed6a618b09b30241fe1c2e1e